contracts/permissionless/OrderbookReserve.sol

```diff
@@ -0,0 +1,870 @@
+            (, , , prevId,) = tokenToEthList.getOrderDetails(orderId);
+        }
+
+        return prevId;
+    }
+
+    function getUpdateOrderHintEthToToken(uint32 orderId, uint128 srcAmount, uint128 dstAmount)
+        public
+        view
+        returns (uint32)
+    {
+        require(dstAmount >= minNewOrderSizeWei);
+        uint32 prevId = ethToTokenList.findPrevOrderId(srcAmount, dstAmount);
+
+        if (prevId == orderId) {
+            (,,, prevId,) = ethToTokenList.getOrderDetails(orderId);
+        }
+
+        return prevId;
+    }
+
+    function getTokenToEthOrder(uint32 orderId)
+        public view
+        returns (
+            address _maker,
+            uint128 _srcAmount,
+            uint128 _dstAmount,
+            uint32 _prevId,
+            uint32 _nextId
+        )
+    {
+        return tokenToEthList.getOrderDetails(orderId);
+    }
+
+    function getEthToTokenOrder(uint32 orderId)
+        public view
+        returns (
+            address _maker,
+            uint128 _srcAmount,
+            uint128 _dstAmount,
+            uint32 _prevId,
+            uint32 _nextId
+        )
+    {
+        return ethToTokenList.getOrderDetails(orderId);
+    }
+
+    function calcKncStake(int weiAmount) public view returns(int) {
+        return(weiAmount * int(kncStakePerEtherBps) / 1000);
+    }
+
+    function calcBurnAmount(uint weiAmount) public view returns(uint) {
+        return(weiAmount * makerBurnFeeBps * feeBurnerContract.kncPerEthRatePrecision() / (1000 * PRECISION));
+    }
+
+    function makerUnusedKNC(address maker) public view returns (uint) {
+        return (uint(makerKnc[maker].freeKnc));
+    }
+
+    function makerStakedKNC(address maker) public view returns (uint) {
+        return (uint(makerKnc[maker].kncOnStake));
+    }
+
+    function getEthToTokenOrderList() public view returns(uint32[] orderList) {
+
+        OrderListInterface list = ethToTokenList;
+        return getList(list);
+    }
+
+    function getTokenToEthOrderList() public view returns(uint32[] orderList) {
+
+        OrderListInterface list = tokenToEthList;
+        return getList(list);
+    }
+
+    function getList(OrderListInterface list) internal view returns(uint32[] memory orderList) {
+        OrderData memory orderData;
+
+        uint32 orderId;
+        bool isEmpty;
+
+        (orderId, isEmpty) = list.getFirstOrder();
+        if (isEmpty) return(new uint32[](0));
+
+        uint numOrders = 0;
+
+        for (; !orderData.isLastOrder; orderId = orderData.nextId) {
+            orderData = getOrderData(list, orderId);
+            numOrders++;
+        }
+
+        orderList = new uint32[](numOrders);
+
+        (orderId, orderData.isLastOrder) = list.getFirstOrder();
+
+        for (uint i = 0; i < numOrders; i++) {
+            orderList[i] = orderId;
+            orderData = getOrderData(list, orderId);
+            orderId = orderData.nextId;
+        }
+    }
+
+    event NewLimitOrder(
```

Comment:
maybe just newOrder? the word "limit" only appears in this event name and might b confusing. Limit is more suitable when a user looks at an existing orderbook.

Reply:
not sure.
its a limit order.

consult Yaron maybe. was his idea.